src/pybind/mgr/cephadm/module.py

```diff
@@ -417,6 +433,9 @@ def __init__(self, *args: Any, **kwargs: Any):
             'node-exporter': self.node_exporter_service,
             'crash': self.crash_service,
             'iscsi': self.iscsi_service,
+            'ha-rgw': self.ha_rgw_service,
+            'haproxy': self.haproxy_service,
+            'keepalived': self.keepalived_service,
```

Comment:
`haproxy` and `keepalived` are no service types, but daemon types. Thus we should never call something like 

```python
self.cephadm_services['keepalived']
```
or we have a big problem keeping sanity here.